Matrix.java

```diff
@@ -0,0 +1,210 @@
+class Matrix
+{
+    final double[][] a;
+    
+    Matrix(double[][] A)
+    {
+	this(A,0,0);
+    }
+    Matrix(double[][] A, int m, int n)
```

Comment:
I agree. Although I would go a little more in depth with the auto-generated Javadoc stubs that Eclipse uses of the form:
`/**
* method_name -Method descriptor.
* @param parameter -Parameter descriptor.
* @return var -Return var descriptor.
**/`